README.md

```diff
@@ -24,39 +26,63 @@ To install Go, visit [this link](https://golang.org/dl/).
 ## Usage
 Before using MSAL Go, you will need to [register your application with the Microsoft identity platform](https://docs.microsoft.com/azure/active-directory/develop/quickstart-v2-register-an-app).
 
+### Public Surface
+
+The Public API of the library can be found in the following directories under `apps`.
+
+```
+apps/ - Contains all our code
+  confidential/ - The confidential application API
+  public/ - The public application API
+  cache/ - The cache interface that can be implemented to provide persistence cache storage of credentials
+```
+
 Acquiring tokens with MSAL Go follows this general three step pattern. There might be some slight differences for other token acquisition flows. Here is a basic example:
+
 1. MSAL separates [public and confidential client applications](https://tools.ietf.org/html/rfc6749#section-2.1). So, you would create an instance of a `PublicClientApplication` and `ConfidentialClientApplication` and use this throughout the lifetime of your application.
-```go
-publicClientApp, err := msal.CreatePublicClientApplication("your_client_id",
-    "https://login.microsoftonline.com/Enter_the_Tenant_Name_Here")
-```
+   * Initializing a public client:
 
-2. MSAL comes packaged with an in-memory cache. Utilizing the cache is optional, but we would highly recommend it.
-```go
-accounts := publicClientApp.GetAccounts()
-// Assuming the user wanted the first account
-userAccount := accounts[0]
-scopes := []string{"your_scope"}
-silentParams := msal.CreateAcquireTokenSilentParametersWithAccount(scopes, userAccount)
-result, err := publicClientApp.AcquireTokenSilent(silentParams)
-```
+    ```go
+    app, err := public.New("client_id", public.WithAuthority("https://login.microsoftonline.com/Enter_The_Tenant_Name_Here"))
```

Comment:
>app [](start = 4, length = 3)

I think this should be named publicClientApp like before, any reason to change?

Reply:
I picked the code snippet from the updated samples. I think `publicClientApp` should be fine too. Although in one of my PRs earlier we shortened something like a `labApplication` to just `app`. Is it a Go convention to have shorter names ? @keegan-caruso , @element-of-surprise ?

Reply:
I still think readable code needs names that indicates what the stuff is for #cleancode :)

---
In reply to: [562163645](https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/146#discussion_r562163645) [](ancestors = 562163645)

Reply:
Agreed, updated in the latest [commit](https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/146/commits/4179b9ba833b42c58ed822ee1f09c2138f1175a6)